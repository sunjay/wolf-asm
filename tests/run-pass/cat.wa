section .code

main:
  push $fp
  mov $fp, $sp

loop:
  # Loop through and write each received byte
  load1 $0, 0xffff_0004

  # Quit at EOF
  cmp $0, 0
  jz end

  # Write the character
  store1 0xffff_000c, $0

  # Continue the loop
  jmp loop

end:
  pop $fp
  ret
