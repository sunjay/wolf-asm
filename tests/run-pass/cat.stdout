
                        Enter ARIEL

  ARIEL. All hail, great master! grave sir, hail! I come
    To answer thy best pleasure; be't to fly,
    To swim, to dive into the fire, to ride
    On the curl'd clouds. To thy strong bidding task
    Ariel and all his quality.
  PROSPERO. Hast thou, spirit,
    Perform'd to point the tempest that I bade thee?
  ARIEL. To every article.
    I boarded the King's ship; now on the beak,
    Now in the waist, the deck, in every cabin,
    I flam'd amazement. Sometime I'd divide,
    And burn in many places; on the topmast,
    The yards, and bowsprit, would I flame distinctly,
    Then meet and join Jove's lightning, the precursors
    O' th' dreadful thunder-claps, more momentary
    And sight-outrunning were not; the fire and cracks
    Of sulphurous roaring the most mighty Neptune
    Seem to besiege, and make his bold waves tremble,
    Yea, his dread trident shake.
  PROSPERO. My brave spirit!
    Who was so firm, so constant, that this coil
    Would not infect his reason?
  ARIEL. Not a soul
    But felt a fever of the mad, and play'd
    Some tricks of desperation. All but mariners
    Plung'd in the foaming brine, and quit the vessel,
    Then all afire with me; the King's son, Ferdinand,
    With hair up-staring-then like reeds, not hair-
    Was the first man that leapt; cried 'Hell is empty,
    And all the devils are here.'
